packages/charting/src/components/BenchmarkBarChart/BenchmarkBarChart.base.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+
+import { mergeStyles, FocusZone } from 'office-ui-fabric-react';
+import { IProcessedStyleSet, DefaultPalette, IStyle } from 'office-ui-fabric-react/lib/Styling';
+import { classNamesFunction } from 'office-ui-fabric-react/lib/Utilities';
+
+import { IBenchmarkBarChartProps, IBenchmarkBarChartStyleProps, IBenchmarkBarChartStyle } from './BenchmarkBarChart.types';
+
+const getClassNames = classNamesFunction<IBenchmarkBarChartStyleProps, IBenchmarkBarChartStyle>();
+
+export class BenchmarkBarChartBase extends React.Component<IBenchmarkBarChartProps> {
+  private _classNames: IProcessedStyleSet<IBenchmarkBarChartStyle>;
+
+  constructor(props: IBenchmarkBarChartProps) {
+    super(props);
+  }
+
+  public render(): JSX.Element {
+    const { chartTitle, benchmarkLabel, data, benchmarkData, totalData } = this.props.data;
+
+    const dataRatio = Math.round((data / totalData) * 100);
+    const benchmarkRatio = Math.round(((benchmarkData ? benchmarkData : 0) / totalData) * 100);
+
+    let percentageContent = data + '/' + totalData;
+    if (this.props.isPercentage) {
+      percentageContent = dataRatio + '%';
+    }
+
+    let benchmarkAriaLabel = benchmarkLabel || '';
+    benchmarkAriaLabel += benchmarkData;
+
+    const benchmarkStyles: IStyle = {
+      marginLeft: 'calc(' + benchmarkRatio + '% - 8px)'
+    };
+
+    const barChartColor = this.props.color || DefaultPalette.purple;
+    const barHeight = this.props.height || 10;
+
+    const { theme, width, benchmarkColor, styles } = this.props;
+    this._classNames = getClassNames(styles!, {
+      theme: theme!,
+      width: width,
+      benchmarkColor: benchmarkColor
+    });
+
+    return (
+      <div className={this._classNames.root}>
+        <div className={this._classNames.chartHeader}>
+          <FocusZone className={this._classNames.chartTitle}>
+            <strong data-is-focusable={true}>{chartTitle}</strong>
+          </FocusZone>
+          <FocusZone className={this._classNames.percentage}>
+            <strong data-is-focusable={true}>{percentageContent}</strong>
+          </FocusZone>
+        </div>
+        <div className={this._classNames.chartBody}>
+          <FocusZone>
+            <div
+              className={mergeStyles(this._classNames.triangle, benchmarkStyles)}
+              aria-label={benchmarkAriaLabel}
+              data-is-focusable={true}
+            />
+          </FocusZone>
+          <svg className={this._classNames.barChart} height={barHeight}>
```

Comment:
This is where I think you should simply be able to compose in a bar chart. If you need slightly different styling you should expose the props of the barchart in this component's interface. From there you could also expose helper props on this component that you merge in just to make it a little easier. I feel very strongly that this is the best approach here and that building another version of the same chart within the same library is not a path we want to go down.

Reply:
@dzearing for additional thoughts

Reply:
In looking at the bar chart I can see why you might not have wanted to use it but I think it's just an opportunity to look at that component and figure out if there's a good base we can build there. For example, you could expose a render prop in that component to maybe add this content in and save a lot of the extra work. Maybe create a component that is only the bars and hover card and nothing (I think I prefer this approach). I think your component probably has some better accessibility as it looks like you were building with the intent to support keyboard which is awesome. I don't know that the original has that in there so it would be great to get some shared value from that as well if possible

Reply:
Additional guidance from the Fabric wiki on composition. https://github.com/OfficeDev/office-ui-fabric-react/wiki/Component-Design

I don't want to block here but I think for us to be set up for success in the future we should take a look here at what might be best both for this component and the HorizontalBarChart (which I think needs work to support the right model)

Reply:
Thanks for your great suggestion. I will compare these two components to see if there is a better code solution. Currently our website development relies on this component, so we hope to use this component as soon as possible without blocking the existing progress.